CONTRIBUTING.md

```diff
@@ -0,0 +1,176 @@
+4. Create a [branch](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/) of this repo on your own fork where all changes will be made
+5. [Install](#installation) BEEP with the developer options.
+6. Test if your installation worked. `pytest beep`.
+
+You now have everything you need to start making changes!
+
+### B. Writing your code
+
+7. BEEP is developed in [Python](https://en.wikipedia.org/wiki/Python_(programming_language). Make sure to follow our [coding style guidelines](#coding-style-guidelines).
+8. Commit your changes to your branch with [useful, descriptive commit messages](https://chris.beams.io/posts/git-commit/). While developing, you can keep using the GitHub issue you're working on as a place for discussion. [Refer to your commits](https://stackoverflow.com/questions/8910271/how-can-i-reference-a-commit-in-an-issue-comment-on-github) when discussing specific lines of code.
+9. If you want to add a dependency on another library, or re-use code you found somewhere else, have a look at [these guidelines](#dependencies-and-reusing-code).
+
+### C. Merging your changes with BEEP
+
+9. [Test your code!](#testing)
+11. If you added a major new feature, perhaps it should be showcased in an [example notebook](#example-notebooks).
+12. When you feel your code is finished, or at least warrants serious discussion, run the [pre-commit checks](#pre-commit-checks) and then create a [pull request](https://help.github.com/articles/about-pull-requests/).
+13. Once a PR has been created, it will be reviewed by the maintainers. Changes might be suggested which you can make by simply adding new commits to the branch. When everything's finished, someone with the right GitHub permissions will merge your changes into BEEP repository.
+
+
+## Installation
+
+To install BEEP with all developer options, type:
+
+```bash
+pip install -e .
+```
+
+This will
+
+1. Install all the dependencies for BEEP, including the ones for documentation (docs) and development (dev).
+2. Tell Python to use your local beep files when you use `import beep` anywhere on your system.
+
+
+## Coding style guidelines
+
+BEEP follows the [PEP8 recommendations](https://www.python.org/dev/peps/pep-0008/) for coding style. These are very common guidelines, and community tools have been developed to check how well projects implement them.
+
+### Naming
+
+In general, we aim for descriptive class, method, and argument names.
+2. Avoid abbreviations when possible without making names overly long, so `mean` is better than `mu`, but a class name like `MyClass` is fine.
+3. Class names are CamelCase, and start with an upper case letter, for example `FastChargeFeatures` or `ProcessedCyclerRun`. 
+4. Method and variable names are lower case, and use underscores for word separation, for example `x` or `calculate_cycle_life`.
```

Comment:
I think this is generally referred to as snake-case

Reply:
*`snake_case`